### ISSUE TITLE
`ob init` crashes when LANG is unset in the environment

### ISSUE THREAD

**Issue by cdsmith** [obsidiansystems/obelisk#335]
I followed the installation instructions on a fresh Ubuntu 18.04 box, and get this:

    $ ob init
    ob: <stdout>: commitAndReleaseBuffer: invalid argument (invalid character)
    /tmp/ob-init-a473b1789c2100f0/.obelisk/impl: command not cached, building ...
    ob: <stdout>: commitAndReleaseBuffer: invalid argument (invalid character)

**Comment by ryantrinkle** [comment]
Weird! Looks like it [might be iconv-related](https://ghc.haskell.org/trac/ghc/ticket/8118); I'll check it out.

**Comment by cdsmith** [comment]
I tried updating libiconv (part of libc6-dev in Ubuntu), but the latest Ubuntu-packaged version still acts the same.  It looks like there's a workaround by setting an explicit encoding in main, but that looks like it requires rebuilding a bunch of stuff.

**Comment by ryantrinkle** [comment]
Hmm; I just finished installing Ubuntu 18.04.1 (desktop edition, english, x64, minimal install), and `ob init` and `ob run` seem to be working.  Is there anything that might be different about your install?  I'd like to see if I can replicate the issue here.

**Comment by cdsmith** [comment]
It was a fresh VM instance created on Google Cloud Platform from the system image provided by Google, Ubuntu 18.04 on x64.  

Since this is encoding-related, maybe it has more to do with the terminal (I was using the web-based terminal from the GCP console) than the binary?

**Comment by ryantrinkle** [comment]
Ah, that could be!  Unsetting `LANG` seems to reproduce the issue.  We'll get this fixed asap, but in the mean time it looks like `export LANG=en_US.UTF-8` will probably help.

**Comment by ryantrinkle** [comment]
I can also reproduce this on NixOS by unsetting both `LANG` and `LC_CTYPE` (unsetting just one or the other doesn't cause the problem)

**Comment by cdsmith** [comment]
Cool.  That fixes it for me.  Thanks for the help.

**Comment by ryantrinkle** [comment]
@cdsmith No problem!  Glad it's sorted out.  We'll get a proper fix in soon!

**Comment by nomeata** [comment]
I observe the same symptoms, but I do have `LANG` set:
```
~/projekte/programming/kaleidogen/kaleidogen-ob $ export LANG=en_US.UTF-8
~/projekte/programming/kaleidogen/kaleidogen-ob $ ob init
ob: <stdout>: commitAndReleaseBuffer: invalid argument (invalid character)
/tmp/ob-init-5e242d86a0dde4a8/.obelisk/impl: command not cached, building ...
ob: <stdout>: commitAndReleaseBuffer: invalid argument (invalid character)
```
(the last three lines are actually colored, if that matters)

**Comment by hamishmack** [comment]
Data.Text.IO.readFile strikes again.  @snoyberg has some good advice on this one:

> never use Prelude.readFile, Data.Text.IO.readFile, Data.Text.Lazy.IO.readFile, or Data.ByteString.Lazy.readFile. Stick with Data.ByteString.readFile for known-small data, use a streaming package (e.g, conduit) if your choice for large data, and handle the character encoding yourself. And apply this to writeFile and other file-related functions as well.

https://www.snoyman.com/blog/2016/12/beware-of-readfile

**Comment by hamishmack** [comment]
Can someone please confirm that:

    nix-env -f https://github.com/hamishmack/obelisk/archive/explicit-utf8.tar.gz -iA command
    ob init

fixes this issue?

If it does then please merge #355

**Comment by ryantrinkle** [comment]
There is also a potential solution here:
https://github.com/obsidiansystems/obelisk/pull/343

**Comment by nomeata** [comment]
> `nix-env -f https://github.com/hamishmack/obelisk/archive/explicit-utf8.tar.gz -iA command`

fixes it here.

**Comment by ali-abrar** [comment]
Fixed via #343
